.github/workflows/daily.yml

```diff
@@ -0,0 +1,95 @@
+name: Daily Block Run
+
+on:
+  schedule:
+  # At the end of every day
+  - cron: "0 0 * * *"
+
+env:
+  RANGE_SIZE: 50
+  SEQUENCER_REV: 1b1b95cae7ae07b9bc778443ca75ee18008a6bc8
+
+jobs:
+  run-and-compare:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1440
+    env:
+      LLVM_SYS_191_PREFIX: /usr/lib/llvm-19/
+      MLIR_SYS_190_PREFIX: /usr/lib/llvm-19/
+      TABLEGEN_190_PREFIX: /usr/lib/llvm-19/
+      RPC_ENDPOINT_TESTNET: ${{ secrets.RPC_ENDPOINT_TESTNET }}
+      RPC_ENDPOINT_MAINNET: ${{ secrets.RPC_ENDPOINT_MAINNET }}
+    strategy:
+      matrix:
+        block:
+          - 740000
+          - 741000
+          - 742000
+          - 743000
+          - 744000
+          - 745000
+          - 746000
+          - 747000
+          - 748000
+          - 749000
+      fail-fast: false
+    defaults:
+      run:
+        shell: bash
+        working-directory: ./starknet-replay
+
+    steps:
+      # We checkout replay first, as it's the main repository for this workflow
+      - name: Checkout Replay
+        uses: actions/checkout@v4
+        with:
+          repository: lambdaclass/starknet-replay
+          path: starknet-replay
+      # We need native for building the runtime
+      - name: Checkout Native
+        uses: actions/checkout@v4
+        with:
+          path: cairo_native
+
+      # Install dependencies
+      - uses: ./cairo_native/.github/actions/install-linux-deps
+      - name: Setup rust env
+        uses: dtolnay/rust-toolchain@1.80.0
+      - name: Retreive cached dependecies
+        uses: Swatinem/rust-cache@v2
+      - name: Build Cairo Native Runtime Library
+        shell: bash
+        run: |
+          cd ../cairo_native
+          make runtime
+          echo "CAIRO_NATIVE_RUNTIME_LIBRARY=$(pwd)/libcairo_native_runtime.a" > $GITHUB_ENV
+      
+      - name: Patch dependencies
+        run: |
+          # Patches native dependency to local path, to use current cairo native version
+          DEPENDENCY="cairo-native"
+          NEW_PATH="../cairo_native"
+          sed -Ei "s#^($DEPENDENCY *=).*#\1 { path = '$NEW_PATH' }#" Cargo.toml
+          grep $DEPENDENCY Cargo.toml
+
+          # Patches sequencer dependency to specified rev
+          GIT="https://github.com/lambdaclass/sequencer"
+          NEW_REV="$SEQUENCER_REV"
+          sed -Ei "s#(\"$GIT\" *, *rev *= *\").?*(\".*)#\1$NEW_REV\2#" Cargo.toml
+          grep $GIT Cargo.toml
+          
+      - name: Run with Native
+        run: |
+          BLOCK_START=${{ matrix.block }}
+          BLOCK_END=$(($BLOCK_START + $RANGE_SIZE - 1))
+          cargo run --release --features state_dump block-range $BLOCK_START $BLOCK_END mainnet
+        continue-on-error: true
+      - name: Run with VM
+        run: |
+          BLOCK_START=${{ matrix.block }}
+          BLOCK_END=$(($BLOCK_START + $RANGE_SIZE - 1))
+          cargo run --release --features state_dump,only_cairo_vm block-range $BLOCK_START $BLOCK_END mainnet
```

Comment:
Since this is meant to be run during night, I don't really know if this could make a significant impact on the time it takes to complete the ci once we increased the amount of blocks to execute. But it might be a good idea to run both native and the vm at the same time to reduce the execution time, right?.

Reply:
I considered it, but I think is not that simple. As I understand it, we have two possible approaches:

The first approach (used in the `starknet-blocks` workflow) is to have two separate jobs, one uses a matrix to execute different blocks with native and VM, then uploads an artifact with the state dumps. Another job depends on the first one, retreives the artifact, and compares the states. 

The downside with this approach is that if one of the executions fails, then the `compare` job is never ran. This is not desirable, as an RPC error is pretty common when running so many blocks, and would cancel the whole daily execution. There might be a way to work around this, but I don't know how.

This is why I opted for a different approach. Use a matrix, but each job in the matrix is complete and independent from the other ones. A failure in one execution would not cancel the other ones, as they are completely separate. The comparison would not have to wait for the whole execution, as each subcomparison could start as soon as the corresponding execution finishes.

The downside of this is that we cannot run VM and Native at the same time, but I don't think that it matters to much, as we can always increase the amount of jobs in the matrix if we want more parallelism.